### PR TITLE
Review of train_network.py

### PR DESCRIPTION
**Pull request by ayplam**

### CODE REVIEW

File: train_network.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
+import tensorflow as tf
+import numpy as np
+from accuracy import *
```
Comment (ayplam) [inline train_network.py:3]: `import *` not generally recommended as it makes code harder to read and follow. I doubt there are that many things to import in accuracy as well.

File: train_network.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+tf.reset_default_graph()
+directory = os.getcwd()
+
+train_file = os.path.join(directory, 'data', 'train.txt')
+test_file =  os.path.join(directory, 'data', 'val.txt')
+logs = os.path.join(directory, 'logs')
+trainloss = os.path.join(logs, 'train_loss.txt')
+
+
+if os.path.isdir(logs) == False:
```
Comment (ayplam) [inline train_network.py:18]: while `== False` technically gives the same result, `not os.path.isdir(logs)` is more pythonic and appropriate. There's also the difference between using `==` vs `is` 
```
a is None #correct
a == None #incorrect
```

File: train_network.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  os.makedirs(logs)
+
+# choose network, can be either DRN18 or DRN26
+network = 'DRN26'
```
Comment (ayplam) [inline train_network.py:22]: parameters are better off for users to set via `argparse` instead of hardcoding at the top. This is common for scripts but is generally bad practice for writing re-usable/productionizable code or when collaborating on a codebase as a team.

File: train_network.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+# set parameters
+batch_size=8
+num_epochs=100
+use_weights = 1
+num_classes = 5
+image_dims=[500,500,3]
+
+data = DataReader(directory, batch_size, num_epochs, use_weights=1)
```
Comment (ayplam) [inline train_network.py:30]: Why is `use_weights=1`? The variable name suggests that it might be better represented as a boolean (`use_weights=True`)

File: train_network.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+  # check if checkpiont exists
+  ckpt = tf.train.get_checkpoint_state(logs)
+  if ckpt and ckpt.model_checkpoint_path:
+    saver.restore(sess, ckpt.model_checkpoint_path) 
+    print('Restoring session at step {}'.format(global_step.eval()))
+    
+  iteration = global_step.eval()
+  current_epoch = int(np.ceil(iteration/training_iterations)) 
+  
+  while current_epoch < num_epochs:
```
Comment (ayplam) [inline train_network.py:128]: `while` loops are used when you don't know how long the loop should continue for, otherwise `for` loops are appropriate. Also done again on L132. `for i in range(training_iterations)` is a pretty common usage (also done this way on L155, which is odd it's not done here)

File: train_network.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    i = 1
+    while i <= training_iterations:
+      _, l = sess.run([training, loss], feed_dict={handle:training_handle, learning_rate:0.0001, train_network:True})
+      iteration = global_step.eval()
+      i = iteration - ((current_epoch-1)*training_iterations) 
+
+      if iteration % 10 == 0:
+        print('Training loss Epoch {} step {}, {} :{}'.format(current_epoch, iteration, i, l))
+        # write loss to file
+        with open(trainloss, 'a') as f:
```
Comment (ayplam) [inline train_network.py:140]: context managers automatically close things. `f.close()` is unnecessary.

File: train_network.py [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+      i = iteration - ((current_epoch-1)*training_iterations) 
+
+      if iteration % 10 == 0:
+        print('Training loss Epoch {} step {}, {} :{}'.format(current_epoch, iteration, i, l))
+        # write loss to file
+        with open(trainloss, 'a') as f: 
+          f.write("Epoch: {} Step: {} Loss: {}\n".format(current_epoch, iteration, l))
+          f.close()
+
+      if (iteration %250 == 0) and (iteration > 0):
```
Comment (ayplam) [inline train_network.py:144]: Is the `iteration > 0` necessary?

File: train_network.py (hunk 5 above)
Comment (ayplam) [inline train_network.py:128]: I'm not entirely positive about what's happening here, but given that there is a `make_initializable_iterator` and you know the number of epochs/training_iterations, I'm fairly certain that this is a poor use of a `while` loop and that a `for` loop can be used. Same with L132. Also this logic where the epoch/iteration is calculated is super confusing and can likely be simplified.
